infrastructure/src/main/java/org/corfudb/infrastructure/log/LogMetadataRecorder.java

```diff
@@ -0,0 +1,56 @@
+package org.corfudb.infrastructure.log;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.infrastructure.BatchProcessor;
+import org.corfudb.infrastructure.BatchWriterOperation;
+import org.corfudb.runtime.proto.service.CorfuMessage;
+import org.corfudb.util.LambdaUtils;
+
+import java.time.Duration;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+public class LogMetadataRecorder {
+
+    private final BatchProcessor batchProcessor;
+
+    private final ScheduledExecutorService recorder;
+
+    private static final Duration LOG_METADATA_RECORDER_INTERVAL = Duration.ofSeconds(30);
+
+    public LogMetadataRecorder(BatchProcessor batchProcessor) {
+        this.batchProcessor = batchProcessor;
+        this.recorder = Executors.newSingleThreadScheduledExecutor(
+            new ThreadFactoryBuilder()
+                .setDaemon(true)
+                .setNameFormat("LogMetadataRecorder")
+                .build());
+        start();
+    }
+
+    private void start() {
+        recorder.scheduleAtFixedRate(
+            () -> LambdaUtils.runSansThrow(this::triggerDump),
+            LogMetadataRecorder.LOG_METADATA_RECORDER_INTERVAL.toMillis() / 2,
+            LogMetadataRecorder.LOG_METADATA_RECORDER_INTERVAL.toMillis(),
+            TimeUnit.MILLISECONDS
+        );
+    }
```

Comment:
I think the periodicity should be based on the number of writes (i.e., logical) and not real time. That way you have a strong upper bound on how much state is not "synced".

Reply:
Will address this comment in a separate PR. At present will mostly focus on the correctness.